HTMLSerializer.js

```diff
@@ -117,12 +137,49 @@ var HTMLSerializer = class {
    */ 
   processDocument(doc) {
     this.html.push('<!DOCTYPE html>\n');
+    this.html.push(''); // Entry where pseudo element style tag will go.
     var nodes = doc.childNodes;
     for (var i = 0, node; node = nodes[i]; i++) {
       if (node.nodeType != Node.DOCUMENT_TYPE_NODE) {
         this.processTree(node, 0);
       }
     }
+    var pseudoElements = `<style>${this.pseudoElementCSS.join('')}</style>`;
+    this.html[1] = pseudoElements;
```

Comment:
Can you assign this '1' to a local variable with a name? It's clear to me now, but I can imagine missing it in a refactoring in the future. Maybe even "var stylePlaceholderIndex = this.html.length;" on line 141?